vault/identity_store_util.go

```diff
@@ -332,7 +332,7 @@ func (i *IdentityStore) upsertEntityInTxn(ctx context.Context, txn *memdb.Txn, e
 			// Otherwise it's still tied to previousEntity and fall through
 			// into merging
 			fallthrough
-		default:
+		case persist && !i.core.ReplicationState().HasState(consts.ReplicationPerformanceSecondary):
```

Comment:
Should this be checking for performance standby too?

Reply:
Fixed.